### ISSUE TITLE
Making the last window available

### ISSUE THREAD

**Issue by WydD** [poetix/protonpack#17]
`StreamUtils.windowed(Stream.of(1,2,3), 2, 2)`
* `[1,2]` only

It would be good to have an option to make the last window available (to avoid data lost). Something like
`StreamUtils.windowed(Stream.of(1,2,3), 2, 2, true)`
* `[1,2]`, `[3]`

From what I saw this is due to the following block in WindowedSpliterator
```java
        if (next.size() != windowSize) {
            next.clear();
        }
```

Thanks

**Comment by poetix** [comment]
Seems reasonable. Could you submit a pull request?

**Comment by poetix** [comment]
Will close when I've performed a release incorporating this change.

**Comment by poetix** [comment]
Released in 1.8
